assets/js/type-defs/cart.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * @typedef {Object} CartTotalItem
```

Comment:
Minor, I'd name the folder `typedefs` for consistency, or just `types`. Not a blocker :)